internal/mempool/mempool.go

```diff
@@ -420,6 +420,7 @@ func (txmp *TxMempool) Update(
 	execTxResult []*abci.ExecTxResult,
 	newPreFn PreCheckFunc,
 	newPostFn PostCheckFunc,
+	recheck bool,
```

Comment:
Might it make more sense to add this to the constructor of the mempool, so that it is available to all methods throughtout its lifecycle?

Reply:
But it can change from height to height as the consensus params change. I don't think any other method requires it hence I thought it made the most sense as an argument whenever we `ApplyBlock`

Reply:
Makes sense.

Reply:
Yeah, if we wanted to pass the mempool the consensus params every height, we could easily modify this method in the future